libstuff/libstuff.cpp

```diff
@@ -2840,6 +2840,31 @@ void SLogLevel(int level) {
     setlogmask(_g_SLogMask);
 }
 
+void SLogAtLevel(int level, const string& message) {
```

Comment:
This function call sort of breaks logging for everything that uses it, in that any call to `SLogAtLevel` will log a line like this:
```
2024-04-24T20:59:37.868658+00:00 db1.sjc bedrock: xxxxxx (libstuff.cpp:2848) _changeState [main] [info] {auth.db1.sjc/LEADING} Switching from 'LEADING' to 'STANDINGDOWN'
```

Notably, the logged file and line will always be in libstuff.cpp inside `SLogAtLevel`.

I think we should move the `switch` block below directly into SLogStackTrace since that's the only place this is used.

Reply:
Ah wow great catch thanks. Will do, and I'll add this check to the test steps.

Reply:
I moved this into the catch and tested, and I see that the stack trace lines still always have the file and line in SLog.cpp. I think that makes sense because the line isn't available in each frame of the call stack. So I won't make this a requirement in the test steps.

I believe you were saying that calling SLogAtLevel directly would overwrite the proper file and line, and therefore it's not a valuable function.

```
2024-04-25T17:16:25.468833+00:00 expensidev2004 bedrock: zNn9yx (SLog.cpp:15) SLogStackTrace [socket259] [dbug]
2024-04-25T17:16:25.468870+00:00 expensidev2004 bedrock: zNn9yx (SLog.cpp:15) SLogStackTrace [socket259] [dbug] SLogStackTrace(int) [0xaaaab29f4af8]
2024-04-25T17:16:25.468917+00:00 expensidev2004 bedrock: zNn9yx (SLog.cpp:15) SLogStackTrace [socket259] [dbug] Report::setNameValuePair(SQLite&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::map<long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<long>, std::allocator<std::pair<long const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&) [0xffff7a8f8458]
2024-04-25T17:16:25.468945+00:00 expensidev2004 bedrock: zNn9yx (SLog.cpp:15) SLogStackTrace [socket259] [dbug] Report::setNameValuePair(SQLite&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) [0xffff7a8fa08c]
2024-04-25T17:16:25.468968+00:00 expensidev2004 bedrock: zNn9yx (SLog.cpp:15) SLogStackTrace [socket259] [dbug] Report::getOrCreateIOUReport(SQLite&, long, long, long, bool, long, long) [0xffff7a9313b4]
```